packages/public/inspection-capture-web/src/hooks/useSightState.ts

```diff
@@ -0,0 +1,40 @@
+import { Sight } from '@monkvision/types';
+import { useState } from 'react';
+
+/**
+ * Custom hook used to initialize and manipulate sight state.
+ */
+export function useSightState(sights: Sight[]) {
+  const [sightSelected, setSightSelected] = useState<Sight>(sights[0]);
+  const [sightsTaken, setSightsTaken] = useState<Sight[]>([]);
+
+  const handleSightSelected = (sight: Sight): void => {
+    setSightSelected(sight);
+  };
+
+  const handleSightTaken = (): void => {
+    if (!sightSelected || sightsTaken.includes(sightSelected)) {
+      return;
+    }
+
+    setSightsTaken((prevSightsTaken) => {
+      const updatedSightsTaken = [...prevSightsTaken, sightSelected];
+
+      const sightsNotTaken = sights.filter((sight) => !updatedSightsTaken.includes(sight));
+      const nextSight = sightsNotTaken[0];
+
+      if (nextSight) {
+        handleSightSelected(nextSight);
+      }
+
+      return updatedSightsTaken;
+    });
+  };
```

Comment:
Hmm I don't really like calling `setState` inside another `setState` callback. It gets messy and it's hard to read. I think. I think you could change this function to :

```typescript
const handleSightTaken = () => {
  setSightsTaken((value) => [...value, selectedSight]);
  const nextSight = sights.filter((sight) => !updatedSightsTaken.includes(sight))[0];
  if (nextSight) {
    setSelectedSight(nextSight);
  }
};
```

Also one thing to note : you don't have to specify the return type when declaring arrow functions, I think it is overkill

Reply:
yeah I agree, lack of readability! 

So I found an other way to update `SelectedSight` state after being sure that `setSightsTaken` have really updated his state:
```
useEffect(() => {
    const nextSight = sights.filter((sight) => !sightsTaken.includes(sight))[0];
    if (nextSight) {
      setSelectedSight(nextSight);
    }
  }, [sightsTaken]);
```
Don't you think that `useSelectedSight` hook is now a bit useless ? Should I move the code to `PhotoCaptureHUD` which is the only place that we use it?

Reply:
Hmm I don't understand why the first solution I suggested isn't good enough? Having a random useEffect in the component that could simply be replaced by a few lines of codes in the handler function isn't really optimal so could you explain to me why you need this?

Reply:
```typescript
const handleSightTaken = () => {
  const updatedSightsTaken = [...sightsTaken, selectedSight];
  setSightsTaken(updatedSightsTaken);
  const nextSight = sights.filter((sight) => !updatedSightsTaken.includes(sight))[0];
  if (nextSight) {
    setSelectedSight(nextSight);
  }
};
```